### DISCUSSION TITLE
Git commit

### DISCUSSION THREAD

**Discussion by CyberByte-04** [community/community#136826] [Actions] [answered]
### Select Topic Area

Question

### Body

How can I revert the last commit in Git without losing the changes?

**Comment by DarkPhoenix-04** [comment, marked as answer]
Use git reset --soft HEAD~1 to undo the last commit while keeping your changes in the working directory.
Your files will still have the changes, but they will be unstaged. You can now make further modifications if needed and commit again using git commit -m "your message".
